cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -358,17 +364,70 @@ func (p *GitHubCiProvider) configureConnection(
 	repoDetails *gitRepositoryDetails,
 	infraOptions provisioning.Options,
 	credentials json.RawMessage,
+	authType PipelineAuthType,
 	console input.Console) error {
 
+	// Oidc + Terraform is not a supported combination
+	if infraOptions.Provider == provisioning.Terraform {
+		// Throw error if Oidc is explicitly requested
+		if authType == AuthTypeOidc {
+			return fmt.Errorf(
+				"Terraform does not support OIDC authentication. Service Principal with client ID and client secret must be used. %w",
+				ErrAuthNotSupported,
+			)
+		}
```

Comment:
We can move this validation to `(p *GitHubScmProvider) preConfigureCheck` which is run even before setting up the SCM or anything.

`preConfigureCheck()` runs before `CreateOrUpdateServicePrincipal`.   Let's not create the principal if we know there's a configuration error